internal/graph/check_fast_path.go

```diff
@@ -598,6 +600,9 @@ func (c *LocalChecker) checkTTUFastPathV2(ctx context.Context, req *ResolveCheck
 	return c.resolveFastPath(ctx, leftChans, storage.WrapIterator(storage.TTUKind, iter))
 }
 
+// fanInIteratorChannels aggregates the messages coming from the input channels into a single output channel.
+// It expects the input channels to be closed.
```

Comment:
I have not spent time investigating whether this is the case. In the test, if you don't close the input channel, the test stalls.

Reply:
Sorry, i meant that i didn't investigate whether every place where this function is called is closing the input channels.

Reply:
Updated as suggested